armi/bookkeeping/db/database3.py

```diff
@@ -1298,15 +1307,29 @@ def _packLocation(loc: grids.LocationBase) -> Tuple[str, List]:
     for everything else.
 
     Shrink grid locator names for storage efficiency.
+    
+    Notes
+    -----
+    Contains some conditionals to still load databases made before
+    db version 3.3 which can be removed once no users care about
+    those DBs anymore.
     """
+    oldStyle = version[0] == 3 and version[1] < 3
+    if oldStyle:
+        locationType = loc.__class__.__name__
     if loc is None:
-        locationType = LOC_NONE
+        if oldStyle:
```

Comment:
For versioned stuff like this, I tend to prefer branching a little bit higher -- maybe even to the point of making two separate functions (one for version >=3, the other for version <=2). Doing so makes it less likely that a future change will affect what would otherwise be a perfectly stable implementation for previous versions. In cases like this, one of the primary motivations for DRY (updating one place updates all clients) is the opposite of what we want.